geonames/geonames-home-country.php

```diff
@@ -4,48 +4,45 @@
 include_once("../konfig-tools.php");
 include_once("$geokrety_www/templates/konfig.php");
 require_once "$geokrety_www/__sentry.php";
-
-$link = DBPConnect();
-
-
-/* ----------------------------------------------- kraj ------------------------------------------------------- */
-
 include("$geokrety_www/get_country_from_coords.php");
+include("geonames_api.php");
 
 
+// --- kraj - country - get user without country but with home coordinates
+$userWithoutCountrySql = "SELECT `userid`, `lat`, `lon` FROM `gk-users` WHERE `lat` != '' and `lon` != '' and (country ='' or country is null)";
+$nbUserUpdated=0;
 
-//ok:
-$result = mysqli_query($link, "SELECT `userid`, `lat`, `lon` FROM `gk-users` WHERE `lat` != '' and `lon` != '' and country =''");
+$link = GKDB::getLink();
+$result = mysqli_query($link,$userWithoutCountrySql);
+if ($result === FALSE) {
+  echo 'invalid query:'.mysqli_error($link).'\n';
+  die;
+}
 
 while ($row = mysqli_fetch_array($result)) {
     list($userid, $lat, $lon) = $row;
-
-
     $country='';
-
     echo "$userid, $lat, $lon::  ";
-
+    // first try from geo.kumy.org
     $country=get_country_from_coords($lat, $lon);
     echo "[ $country] ";
 
     if ($country == '' or $country=='xyz' or $country==null) {
-        $handle = fopen("http://ws.geonames.org/countryCode?lat=$lat&lng=$lon&radius=30", "rb");
-        $country = strtolower(trim(fread($handle, 3)));
-        echo "geonames...";
+        // in case of failure, ask to geonames
+        $country = geonames_country_from_coords($geonamesEndpoint, $geonamesUsername, $lat, $lon);
         if (($country == '<!d') or ($country == '')) {
             $country = 'xyz';
             echo "bach!\n";
         }
-        fclose($handle);
         sleep(2);
     }
-
-
     echo "$country\n";
 
+    // update user with country
     $sql = "UPDATE `gk-users` SET  `country`='$country' WHERE `userid` = '$userid' LIMIT 1";
     $result2 = mysqli_query($link, $sql);
+    $nbUserUpdated++;
```

Comment:
:+1:

BTW It would be awesome if we could retrieve/publish such information as [prometheus format](https://prometheus.io/docs/instrumenting/clientlibs/) :1st_place_medal:

Reply:
you mean use 'in memory' Prometheus client?
https://github.com/Jimdo/prometheus_client_php/blob/master/README.md

a counter for each update? plus an endpoint to render them?

btw, I don't found this script in cron, are they manually launched?

Reply:
> btw, I don't found this script in cron, are they manually launched?

probably a mistake, I don't launch anything manually

> you mean use 'in memory' Prometheus client?

I mean I imagine our site can expose some metrics via the /metrics endpoint (number of geokrety, number of moves, number of user, cron counts or any other measurable value). So they can be collected by prometheus. I've only used such endpoints but never develop some client. So can;t tell you if it 'in memory' or any other thing to use.

Reply:
-cron: oky theres is ruchy and waypoint but no home(so I will add)
-metrics: ok thx